File: 2021/5/1.py
from shared import parse_line, record_coordinates, count_overlap_points

coord_map = {}

with open("input.txt") as f:
    for line in f:
        x1, y1, x2, y2 = parse_line(line)

        if x1 != x2 and y1 != y2:
            continue

        record_coordinates(coord_map, x1, x2, y1, y2)

print(count_overlap_points(coord_map))

File: 2022/1/1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	f, _ := os.Open("input.txt")

	defer f.Close()

	scanner := bufio.NewScanner(f)
	maxCalories := 0
	currCalories := 0

	for scanner.Scan() {
		calories, err := strconv.Atoi(scanner.Text())

		// Empty line
		if err != nil {
			if currCalories > maxCalories {
				maxCalories = currCalories
			}
			currCalories = 0
		} else {
			currCalories += calories
		}
	}

	fmt.Printf("Maximum calories is: %d", maxCalories)
}
